### PR TITLE
Remove Element.setStatus(), various other fixes/nitpicks via Cassandra bump

### PR DESCRIPTION
**Pull request by nickbp**
Was also seeing local errors about a missing `LIBMESOS_URI` which are now fixed.

### CODE REVIEW

File: sdk/scheduler/src/main/java/org/apache/mesos/scheduler/plan/DefaultStep.java [hunk 1]
```diff
@@ -137,16 +105,18 @@ public Status getStatus() {
     }
 
     @Override
-    public List<Element> getChildren() {
+    public List<Element<?>> getChildren() {
```
Comment (gabrielhartmann) [inline sdk/scheduler/src/main/java/org/apache/mesos/scheduler/plan/DefaultStep.java]: Why change this?

File: sdk/scheduler/src/main/java/org/apache/mesos/scheduler/plan/DefaultStep.java [hunk 2]
```diff
@@ -137,16 +105,18 @@ public Status getStatus() {
...
         return Collections.emptyList();
     }
 
     @Override
-    @SuppressWarnings("unchecked")
-    public Strategy<? extends Element> getStrategy() {
+    public Strategy<Step> getStrategy() {
```
Comment (gabrielhartmann) [inline sdk/scheduler/src/main/java/org/apache/mesos/scheduler/plan/DefaultStep.java]: A Phase has a strategy to deploy Steps.  A Step doesn't have Steps as children.  I suppose this gets rid of the unchecked warning...

File: sdk/scheduler/src/main/java/org/apache/mesos/scheduler/plan/DefaultStep.java (hunk 1 above)
Comment (nickbp) [inline sdk/scheduler/src/main/java/org/apache/mesos/scheduler/plan/DefaultStep.java]: Just fixed a warning in eclipse

File: sdk/scheduler/src/main/java/org/apache/mesos/scheduler/plan/DefaultStep.java (hunk 2 above)
Comment (nickbp) [inline sdk/scheduler/src/main/java/org/apache/mesos/scheduler/plan/DefaultStep.java]: Yes this just was to remove the need for the `SuppressWarnings`

File: sdk/scheduler/src/main/java/org/apache/mesos/scheduler/plan/PlanUtils.java [hunk 3]
```diff
@@ -27,99 +35,135 @@
...
                 element.getClass().getName(), element.getName(), element.getId(), element.getStatus());
     }
 
-    @SuppressWarnings("unchecked")
-    public static final List<String> getErrors(List<String> errors, Element parent) {
+    /**
+     * Returns all of the errors from the parent and all its child {@link Element}s. Intended for
+     * use by implementations of {@Link Element#getErrors()}.
+     *
+     * @param parentErrors Errors to be included from the parent itself, will be modified in-place.
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/org/apache/mesos/scheduler/plan/PlanUtils.java]: Correct, this was once the case until it was fixed. Updated the javadoc

**Comment by nickbp** [comment]
Filed a ticket about refactoring PlanUtils.

**Pull request merged**